index.js

```diff
@@ -99,7 +103,47 @@ async function pay (plugin, {
   }
 }
 
+async function pull (plugin, {
+  pointer,
+  amount,
+  streamOpts = {}
+}) {
+  await plugin.connect()
+  const receiveMax = amount || Infinity
+
+  const response = await query(pointer)
+
+  if (response.contentType.indexOf('application/spsp4+json') !== -1) {
+    const ilpConn = await createConnection({
+      plugin,
+      destinationAccount: response.destinationAccount,
+      sharedSecret: response.sharedSecret,
+      ...streamOpts
+    })
+
+    const stream = await ilpConn.createStream()
+
+    try {
+      await stream.receiveTotal(receiveMax, { timeout: streamOpts.timeout })
+    } catch (err) {
+      const totalReceived = stream.totalReceived
+      await ilpConn.end()
```

Comment:
It looks like if you attempt to pull more than the server's allowed amount and don't specify a shorter timeout for `receiveTotal`, the connection will hit its idle timeout (also 1 minute) and call `destroy` before we call `end` here.
https://github.com/interledgerjs/ilp-protocol-stream/blob/master/src/connection.ts#L1510-L1531
That (at least sometimes) results in an `ilp-plugin-btp` timeout error being thrown while we wait for `end`.
It seems like we should catch any errors thrown by `end` here and/or specify a longer `idleTimeout` for the connection.

Reply:
> That (at least sometimes) results in an `ilp-plugin-btp` timeout error being thrown while we wait for `end`.

Why does that happen? I mean I can catch the error (and then do what?) but I don't understand why that happens (sometimes) in the first place.

Reply:
I'm actually getting this more often now:
```
  ilp-protocol-stream:Client:Connection:info closing connection +0ms
  ilp-protocol-stream:Client:Connection:debug starting send loop +1m
  ilp-protocol-stream:Client:Stream:1:info stream is closing +1m
  ilp-protocol-stream:Client:Stream:1:trace starting another send loop to tell the peer the stream was closed +1m
  ilp-protocol-stream:Client:Stream:1:info stream ended +0ms
  ilp-protocol-stream:Client:Connection:debug removing record of stream 1 +1ms
  ilp-protocol-stream:Client:Connection:trace loadAndSendPacket +2ms
  ilp-protocol-stream:Client:Connection:trace sending packet 7 with source amount: 0: {"sequence":"7","ilpPacketType":12,"prepareAmount":"0","frames":[{"type":16,"name":"StreamClose","streamId":"1","errorCode":1,"errorMessage":""}]}) +0ms
  ilp-protocol-stream:Client:Connection:error Connection timed out due to inactivity, destroying connection +0ms
  ilp-protocol-stream:Client:Connection:error destroying connection with error: Error: Connection timed out due to inactivity
    at Connection.testIdle (/home/brandon/ilp-protocol-spsp/node_modules/ilp-protocol-stream/src/connection.ts:1528:6)
    at Timeout.idleTimer.setTimeout [as _onTimeout] (/home/brandon/ilp-protocol-spsp/node_modules/ilp-protocol-stream/src/connection.ts:1516:41)
    at ontimeout (timers.js:436:11)
    at tryOnTimeout (timers.js:300:5)
    at unrefdHandle (timers.js:520:7)
    at Timer.processTimers (timers.js:222:12) +1ms
err Error: Connection timed out due to inactivity
    at Connection.testIdle (/home/brandon/ilp-protocol-spsp/node_modules/ilp-protocol-stream/src/connection.ts:1528:6)
    at Timeout.idleTimer.setTimeout [as _onTimeout] (/home/brandon/ilp-protocol-spsp/node_modules/ilp-protocol-stream/src/connection.ts:1516:41)
    at ontimeout (timers.js:436:11)
    at tryOnTimeout (timers.js:300:5)
    at unrefdHandle (timers.js:520:7)
    at Timer.processTimers (timers.js:222:12)
```
1. `receiveTotal` times out
2. `ilp-protocol-spsp` calls `await ilpConn.end()`
3. STREAM sends a `"StreamClose"` frame
4. Before the response packet is received, the idle timer expires and STREAM throws the `Connection timed out due to inactivity` error. (The connection's idle timer only resets (`bumpIdle`) when a packet is received)

We could either catch and do nothing or log (probably with `ilp-logger`). Either way we want to still throw the `PaymentError` with the `totalReceived`.

Reply:
Thanks for the explanation. I'll try to reproduce it and add your suggestions (i.e. catch and logging)